src/arviz_plots/plots/utils.py

```diff
@@ -224,30 +285,32 @@ def add_reference_lines(
         plot_kwargs = {}
     if aes_map is None:
         aes_map = {}
-    if data_vars is None:
-        data_vars = plot_collection.data.data_vars
+    if sample_dims is None:
+        sample_dims = rcParams["data.sample_dims"]
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
 
     plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
 
     plot_func = vlines if orientation == "vertical" else hlines
 
-    _, ref_aes, ref_ignore = filter_aes(plot_collection, aes_map, "reference", "sample")
+    _, ref_aes, ref_ignore = filter_aes(plot_collection, aes_map, "reference", sample_dims)
```

Comment:
I would use `ref_line` and then also allow a 2nd key in aes_map and plot_kwargs `ref_text` (with default for text to now show it and also can be added in a follow-up PR).